src/engine/CrashLogger.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Godot;
+
+/// <summary>
+///   Crash logger for any unhandled Mono exceptions
+/// </summary>
+public class CrashLogger
+{
+    public static void UnhandledException(object sender, UnhandledExceptionArgs args)
+    {
+        GD.PrintErr("------------ Begin of Crash Log ------------\n" +
```

Comment:
Don't concatenate multiple strings with +, it's very inefficient in C#. Instead I think this could use just commas to separate the parts, if you don't want to use a string builder here.